Userland/Libraries/LibWeb/Layout/TreeBuilder.cpp

```diff
@@ -370,15 +371,68 @@ ErrorOr<void> TreeBuilder::create_layout_tree(DOM::Node& dom_node, TreeBuilder::
         pop_parent();
     }
 
+    if (is<HTML::HTMLMeterElement>(dom_node)) {
```

Comment:
Constructing the inner parts of this in TreeBuilder feels like the wrong approach to me. I think it should be done using a shadow tree inside the HTMLMeterElement. For an example, see [`HTMLInputElement`](https://github.com/SerenityOS/serenity/blob/master/Userland/Libraries/LibWeb/HTML/HTMLInputElement.cpp#L535).